source/extensions/filters/network/mysql_proxy/mysql_config.cc

```diff
@@ -24,13 +30,43 @@ NetworkFilters::MySQLProxy::MySQLConfigFactory::createFilterFactoryFromProtoType
     Server::Configuration::FactoryContext& context) {
 
   ASSERT(!proto_config.stat_prefix().empty());
-
   const std::string stat_prefix = fmt::format("mysql.{}", proto_config.stat_prefix());
 
   MySQLFilterConfigSharedPtr filter_config(
       std::make_shared<MySQLFilterConfig>(stat_prefix, context.scope()));
-  return [filter_config](Network::FilterManager& filter_manager) -> void {
-    filter_manager.addFilter(std::make_shared<MySQLFilter>(filter_config));
+
+  if (!proto_config.has_database_routes()) {
+    return [filter_config](Network::FilterManager& filter_manager) -> void {
+      filter_manager.addFilter(
+          std::make_shared<MySQLMonitorFilter>(filter_config, DecoderFactoryImpl::instance_));
+    };
+  }
+
+  absl::flat_hash_map<std::string, RouteSharedPtr> routes;
+  RouteSharedPtr catch_all_route = nullptr;
+  if (proto_config.has_database_routes()) {
```

Comment:
You do not need to check if `has_database_routes()` returns true, because 7 lines above the method exits when it returns false.

Reply:
fixed.